src/Workspaces/Core/Portable/CodeStyle/CodeStyleOption.cs

```diff
@@ -48,7 +50,11 @@ public CodeStyleOption(T value, NotificationOption notification)
 
         private int EnumValueAsInt32 => (int)(object)Value;
 
-        public NotificationOption Notification { get; set; }
+        public NotificationOption Notification
```

Comment:
are there any setters of this in Roslyn itself?

Reply:
No, no setters internally within the project, but it's listed in this file:
https://github.com/dotnet/roslyn/blob/97b75fcb5807881b0e768357b82c1a3476a3ec2c/src/Workspaces/Core/Portable/PublicAPI.Shipped.txt#L409

Reply:
I'm not sure if/how these types and members are consumed outside of Roslyn.

Reply:
@jasonmalinowski Thoughts here?  Worth breaking this and going to compat council over this?

Reply:
@castholm You're fine.  And, for the moment, i think what you're doing is the safest approach.  Just starting a discussion with stakeholders to see if it's worth doing something about this since you're poking around in this space right now :)

Reply:
I believe it's worth taking this through the compat council not because I think they're going to have any problems with it, but rather implicitly the process is what we need to go through. The value of that process often isn't the actual council asking questions, but that in prepping you asked the questions first. I don't know if Visual Studio for Mac or Omnisharp might potentially call this if they're trying to do something, so we need to check. Also another thing to look: is this bound with WPF two-way binding in Tools > Options?

I haven't fully looked at this PR, but honestly I'm surprised this is even settable and this type is mutable. (I'm worrying WPF binding is the reason for this.) If these are held by OptionSets than thats a really big no-no. This is for some reason making me think either @sharwell or @jmarolf noticed this semi-recently as well...

Reply:
I think @mavasani 	would be the expert on why this was added in the first place, but I am all for making this change

Reply:
@jmarolf This infrastructure has been in place before I joined the IDE team, so unfortunately, I don't have much context on this either.

Reply:
I agree with @CyrusNajmabadi and @jasonmalinowski overall. Doesn't seem clear why we would need to have this setter at all.